security/i2pd/Portfile

```
@@ -0,0 +1,164 @@
+long_description        ${description}
+homepage                https://i2pd.website \
+                        https://i2pd.readthedocs.io
+checksums               rmd160  a5cbcba1ae16d0a12936f1c5699b59d6c72f46ca \
+                        sha256  9061265a167c7f2d98e50736bb44c66c03e8c54ac0ea2fe9a6040c8b23cd5169 \
+                        size    638151
+
+depends_lib-append      port:zlib
+
+patch.pre_args          -p1
+patchfiles-append       0001-Correct-define-for-Darwin-PPC-is-__POWERPC__.patch \
+                        0002-util.cpp-do-not-use-pthread_setname_np-on-unsupporte.patch \
+                        0003-i2pd.conf-adjust-for-MacPorts.patch
+
+set i2pddir             ${prefix}/var/db/${name}
+
+post-patch {
+    reinplace "s|@PREFIX@|${prefix}|g" ${worksrcpath}/contrib/i2pd.conf
+    reinplace "s|@I2PDDIR@|${i2pddir}|g" ${worksrcpath}/contrib/i2pd.conf
+}
+
+set i2puser             _i2p
+add_users               ${i2puser} group=${i2puser} home=${i2pddir}
+
+cmake.source_dir        ${worksrcpath}/build
+
+# Minimum required is C++11, C++17 used if supported.
+compiler.cxx_standard   2011
+
+# https://github.com/PurpleI2P/i2pd/issues/1798
+compiler.blacklist-append \
+                        {clang < 600}
+
+# These are defaults, but better have them explicit:
+configure.args-append   -DWITH_LIBRARY=ON \
+                        -DWITH_BINARY=ON \
+                        -DWITH_STATIC=OFF
+
+# See Makefile.osx
+configure.ldflags-append "-lz -lcrypto -lpthread"
+configure.ldflags-append "-dead_strip -dead_strip_dylibs -bind_at_load"
+
+if {${build_arch} in [list ppc ppc64]} {
+    configure.args-append \
+                        -Dppc_support=TRUE \
+                        -DWITH_AESNI=OFF
+    # Temporary fix, enabling Boost atomics in place of default GCC.
+    # See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107590
+    # https://github.com/PurpleI2P/i2pd/issues/1726
+    # https://github.com/boostorg/smart_ptr/issues/105
+    configure.cxxflags-append \
+                        -DBOOST_SP_USE_STD_ATOMIC
+}
+
+post-destroot {
+    xinstall -d ${destroot}${prefix}/etc/${name}
+    foreach f {i2pd.conf subscriptions.txt tunnels.conf} {
+        if {[file exists ${worksrcpath}/contrib/${f}]} {
+            xinstall -m 0664 ${worksrcpath}/contrib/${f} ${destroot}${prefix}/etc/${name}
+        }
+    }
+    if {[file exists ${worksrcpath}/contrib/tunnels.d]} {
+        copy ${worksrcpath}/contrib/tunnels.d ${destroot}${prefix}/etc/${name}
+    }
+    if {[file exists ${worksrcpath}/contrib/certificates]} {
+        xinstall -d ${destroot}${prefix}/share/${name}
+        copy ${worksrcpath}/contrib/certificates ${destroot}${prefix}/share/${name}
+    }
+    xinstall -d ${destroot}${prefix}/share/doc/${name}
+    foreach f {ChangeLog LICENSE Makefile.osx README.md} {
+        if {[file exists ${worksrcpath}/${f}]} {
+            xinstall -m 0644 ${worksrcpath}/${f} ${destroot}${prefix}/share/doc/${name}
+        }
+    }
```

Why wouldn't these files exist? Same question for the preceding ` foreach` blocks.

Assuming they exist, you can install them all at once with less code:

```suggestion
    xinstall -W ${worksrcpath} -m 0644 ChangeLog LICENSE Makefile.osx README.md \
        ${destroot}${prefix}/share/doc/${name}
```

Do you really want to copy a Makefile into the documentation directory? That's unusual.

---

@ryandesign `Makefile.osx` is not necessary, of course. I thought to keep it simply because `i2pd` supports alternative build system. Can be dropped.

`file exists` condition is a safe-guard against a case when sources are modified and build gonna fail due to some file not existing or renamed. I have had such issues with other ports.
But also can be dropped, if undesirable.